cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -201,22 +207,62 @@ public void configure(Map<String, ?> configs) {
-   * registry does not work with 2.6+. Therefore if the new class exists, we use it and if it doesn't exist we use the
-   * old one.
+   * Starting with Kafka 3.3.0 a new class, "org.apache.kafka.server.metrics.KafkaYammerMetrics", provides the default Metrics Registry.
+   *
+   * This is the third default Metrics Registry class change since Kafka 2.5:
+   *   - Metrics Registry class in Kafka 3.3+: org.apache.kafka.server.metrics.KafkaYammerMetrics
+   *   - Metrics Registry class in Kafka 2.6+: kafka.metrics.KafkaYammerMetrics
+   *   - Metrics Registry class in Kafka 2.5-: com.yammer.metrics.Metrics
+   **
+   * The older default registries do not work with the newer versions of Kafka. Therefore, if the new class exists, we use it and if
+   * it doesn't exist we will fall back on the older ones.
    *
    * Once CC supports only 2.6.0 and newer, we can clean this up and use only KafkaYammerMetrics all the time.
    *
    * @return  MetricsRegistry with Kafka metrics
    */
   private static MetricsRegistry metricsRegistry() {
+    Object metricsRegistry;
+    Class<?> metrics;
+
     try {
-      Class.forName("kafka.metrics.KafkaYammerMetrics");
-      LOG.info("KafkaYammerMetrics found and will be used.");
-      return KafkaYammerMetrics.defaultRegistry();
+      // First we try to get the KafkaYammerMetrics class for Kafka 3.3+
+      metrics = Class.forName(YAMMER_METRICS_IN_KAFKA_3_3_AND_LATER);
+      LOG.info("Found class {} for Kafka 3.3 and newer.", YAMMER_METRICS_IN_KAFKA_3_3_AND_LATER);
     } catch (ClassNotFoundException e) {
-      LOG.info("KafkaYammerMetrics not found. Metrics will be used.");
-      return Metrics.defaultRegistry();
+      LOG.info("Class {} not found. We are probably on Kafka 3.2 or older.", YAMMER_METRICS_IN_KAFKA_3_3_AND_LATER);
+
+      // We did not find the KafkaYammerMetrics class from Kafka 3.3+. So we are probably on older Kafka version
+      //     => we will try the older class for Kafka 2.6+.
+      try {
+        metrics = Class.forName(YAMMER_METRICS_IN_KAFKA_2_6_AND_LATER);
+        LOG.info("Found class {} for Kafka 2.6 and newer.", YAMMER_METRICS_IN_KAFKA_2_6_AND_LATER);
+      } catch (ClassNotFoundException ee) {
+        LOG.info("Class {} not found. We are probably on Kafka 2.5 or older.", YAMMER_METRICS_IN_KAFKA_2_6_AND_LATER);
+
+        // We did not find the KafkaYammerMetrics class from Kafka 2.6+. So we are probably on older Kafka version
+        //     => we will try the older class for Kafka 2.5-.
+        try {
+          metrics = Class.forName(YAMMER_METRICS_IN_KAFKA_2_5_AND_EARLIER);
+          LOG.info("Found class {} for Kafka 2.5 and earlier.", YAMMER_METRICS_IN_KAFKA_2_5_AND_EARLIER);
+        } catch (ClassNotFoundException eee) {
+          // No class was found for any Kafka version => we should fail
+          throw new RuntimeException("Failed to find Yammer Metrics class", eee);
```

Comment:
Is there a reason that we throws exception instead of returning Metrics.defaultRegistry()？ I'm asking because I see the code was returning Metrics.defaultRegistry() before.

Reply:
We do return Metrics.defaultRegistry() by the end of the method if the class and method exists, this makes it consistent with how we would check the defaultRegistry() methods of the other classes

Reply:
Ah I see. Thanks for the explanation.